From df1f43a0d3490faa73962b4c9d9fcc2646e79b31 Mon Sep 17 00:00:00 2001
From: Robin Gong <b38343@freescale.com>
Date: Mon, 12 Mar 2012 18:37:59 +0800
Subject: [PATCH 1155/2463] ENGR00176571 MX6DL: Added support for 1.2GHz ARM
 Frequency

Added the new 1.2GHz working point.Currently 'arm_freq=1200"
should be added to commandline for the core to run at 1.2GHz

Signed-off-by: Robin Gong <B38343@freescale.com>
---
 arch/arm/mach-mx6/cpu_op-mx6.c |   30 ++++++++++++++++++++++++++++--
 1 files changed, 28 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-mx6/cpu_op-mx6.c b/arch/arm/mach-mx6/cpu_op-mx6.c
index 6f03010..088a9fd 100644
--- a/arch/arm/mach-mx6/cpu_op-mx6.c
+++ b/arch/arm/mach-mx6/cpu_op-mx6.c
@@ -97,7 +97,30 @@ static struct cpu_op mx6_cpu_op[] = {
 	 .cpu_voltage = 850000,},
 };
 
-/* working point(wp): 0 - 1GHzMHz; 1 - 800MHz, 3 - 400MHz, 4  - 200MHz */
+/* working point(wp): 0 - 1.2GHz; 1 - 800MHz, 2 - 400MHz, 3  - 200MHz */
+static struct cpu_op mx6dl_cpu_op_1_2G[] = {
+	{
+	 .pll_rate = 1200000000,
+	 .cpu_rate = 1200000000,
+	 .cpu_podf = 0,
+	 .cpu_voltage = 1275000,},
+	{
+	 .pll_rate = 792000000,
+	 .cpu_rate = 792000000,
+	 .cpu_podf = 0,
+	 .cpu_voltage = 1100000,},
+	 {
+	  .pll_rate = 792000000,
+	  .cpu_rate = 396000000,
+	  .cpu_podf = 1,
+	  .cpu_voltage = 1000000,},
+	{
+	 .pll_rate = 792000000,
+	 .cpu_rate = 198000000,
+	 .cpu_podf = 3,
+	 .cpu_voltage = 1000000,},
+};
+/* working point(wp): 0 - 1GHz; 1 - 800MHz, 2 - 400MHz, 3  - 200MHz */
 static struct cpu_op mx6dl_cpu_op_1G[] = {
 	{
 	 .pll_rate = 996000000,
@@ -175,7 +198,10 @@ static struct dvfs_op *mx6_get_dvfs_core_table(int *wp)
 struct cpu_op *mx6_get_cpu_op(int *op)
 {
 	if (cpu_is_mx6dl()) {
-		if (arm_max_freq == CPU_AT_1GHz) {
+		if (arm_max_freq == CPU_AT_1_2GHz) {
+			*op =  num_cpu_op = ARRAY_SIZE(mx6dl_cpu_op_1_2G);
+			return mx6dl_cpu_op_1_2G;
+		} else if (arm_max_freq == CPU_AT_1GHz) {
 			*op =  num_cpu_op = ARRAY_SIZE(mx6dl_cpu_op_1G);
 			return mx6dl_cpu_op_1G;
 		} else {
-- 
1.7.7.4

